### PR TITLE
Generative VQ-VAE Model using PixelCNN for ADNI Dataset 

### PR DESCRIPTION
**Pull request by BigKyleMac**
Hi COMP3710 Teaching team,

This pull request is an implementation of a Generative VQ-VAE Model utilising a PixelCNN (Problem 8) on the ADNI dataset. The VQ-VAE reconstruction images have a mean SSIM of 0.823, and reasonably clear brain images have been generated using PixelCNN.

The pull request contains the following files:

- dataset.py: Contains functions for loading in the data
- modules.py: Contains the classes and components for the VQ-VAE and PixelCNN models
- train.py: File for initialising, compiling, training and saving both the VQ-VAE and PixelCNN models, and for visualising training loss.
- predict.py: File for loading in trained VQ-VAE and PixelCNN models, visualising example VQ-VAE reconstructions, and generating new brain images
- README.md: The README file and report for the project
- images/ : Directory containing images and resources used in README.md

Kind regards,
Kyle Macaskill (45814011)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
- commit messages: OK
- reconstruction: OK
- image gen: OK
- code commenting: OK
- readme: seems: OK

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove .idea files/folder -2
Request Description good

**Comment by BigKyleMac** [comment]
Hi,

Regarding removing the .idea files/folder, I believe I have already done this in the commit "Cleaning files from Pycharm" (2nd most recent one).

In the files changed tab on the pull request, the .idea files are indicated as being deleted now. If this isn't what is meant in the feedback please let me know, but on my end, I cannot see any .idea files to remove.

Thank you,
Kyle
